demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ClericalToolController.scala

```diff
@@ -0,0 +1,174 @@
+package uk.gov.ons.addressIndex.demoui.controllers
+
+import java.util.UUID
+import javax.inject.{Inject, Singleton}
+
+import play.api.Logger
+import play.api.data.Forms._
+import play.api.data._
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, AnyContent, Controller}
+import uk.gov.ons.addressIndex.demoui.client.AddressIndexClientInstance
+import uk.gov.ons.addressIndex.demoui.model._
+import uk.gov.ons.addressIndex.demoui.modules.DemouiConfigModule
+import uk.gov.ons.addressIndex.demoui.utils.ClassHierarchy
+import uk.gov.ons.addressIndex.model.server.response.AddressBySearchResponseContainer
+import uk.gov.ons.addressIndex.model.AddressIndexSearchRequest
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.implicitConversions
+import scala.util.Try
+
+/**
+  * Controller class for a single address to be matched
+  * @param conf
+  * @param messagesApi
+  * @param apiClient
+  * @param ec
+  */
+@Singleton
+class ClericalToolController @Inject()(
+                                       conf : DemouiConfigModule,
+                                       val messagesApi: MessagesApi,
+                                       apiClient: AddressIndexClientInstance,
+                                       classHierarchy: ClassHierarchy
+                                     )(implicit ec: ExecutionContext) extends Controller with I18nSupport {
+
+  val logger = Logger("ClericalToolController")
+  val pageSize = conf.config.limit
+  val maxOff = conf.config.maxOffset
+  val maxPages = (maxOff + pageSize - 1) / pageSize
+
+  /**
+    * Present empty form for user to input address
+    *
+    * @return result to view
+    */
+  def showSingleMatchPage(): Action[AnyContent] = Action.async { implicit request =>
+    logger info ("Clerial Tool: Rendering Single Match Page")
+    val viewToRender = uk.gov.ons.addressIndex.demoui.views.html.clericalTool(
+      singleSearchForm = SingleMatchController.form,
+      warningMessage = None,
+      addressFormat = "paf",
+      pageNum = 1,
+      pageSize = pageSize,
+      pageMax = maxPages,
+      expandRow = -1,
+      addressBySearchResponse = None,
+      classification = None)
+    Future.successful(
+      Ok(viewToRender)
+    )
+  }
+
+  /**
+    * Accept posted form, deal with empty address or pass on to MatchWithInput
+    *
+    * @return result to view or redirect
+    */
+  def doMatch(): Action[AnyContent] = Action.async { implicit request =>
+    val optAddress: Option[String] = Try(request.body.asFormUrlEncoded.get("address").mkString).toOption
+    val addressText = optAddress.getOrElse("")
```

Comment:
`getOrElse` may be applied directly to the `Try` without going through the `Option`